frontend/src/routes/Governance/common/util.tsx

```diff
@@ -463,30 +464,74 @@ export function getPolicySetPolicies(policies: Policy[], policySet: PolicySet) {
   )
 }
 
-export function getPolicyRemediation(policy: Policy | undefined) {
+export function getPolicyRemediation(policy: Policy | undefined, propagatedPolicies: Policy[]) {
   if (!policy) {
     return ''
   }
-  if (policy.spec?.remediationAction) {
-    return policy.spec.remediationAction
-  }
   const templates = policy.spec['policy-templates']
+  const rootRA = policy.spec.remediationAction
+  if (rootRA === REMEDIATION_ACTION.ENFORCE) {
+    return rootRA
+  }
+  // NOTE RA means Remediation Action
   let remediationAggregation = '-'
-  templates?.forEach((template: PolicyTemplate) => {
-    const templateRemediation = template.objectDefinition.spec.remediationAction
-    if (remediationAggregation === 'inform/enforce') {
-      return
-    } else if (remediationAggregation !== '-' && remediationAggregation !== templateRemediation) {
-      remediationAggregation = 'inform/enforce'
-      return
-    } else if (remediationAggregation !== templateRemediation && templateRemediation) {
-      remediationAggregation = templateRemediation
-      return
+  propagatedPolicies.forEach((propaPolicy) => {
+    const propaPolicyRA = propaPolicy.spec.remediationAction
+    if (propaPolicyRA !== rootRA) {
+      if (remediationAggregation === '-') {
+        remediationAggregation = REMEDIATION_ACTION.ENFORCE_OVERRIDDEN
+      } else if (!remediationAggregation.includes(REMEDIATION_ACTION.ENFORCE_OVERRIDDEN)) {
+        remediationAggregation = REMEDIATION_ACTION.INFORM_ENFORCE_OVERRIDEN
+      }
+    } else if (propaPolicyRA != undefined) {
+      if (remediationAggregation === '-') {
+        remediationAggregation = REMEDIATION_ACTION.INFORM
+      } else if (remediationAggregation == REMEDIATION_ACTION.ENFORCE_OVERRIDDEN) {
+        remediationAggregation = REMEDIATION_ACTION.INFORM_ENFORCE_OVERRIDEN
```

Comment:
If you used a `for` loop instead of `.forEach` you could break here to avoid additional iterations for nothing.

Reply:
what you mean additional iterations?

Reply:
I mean that `.forEach` doesn't allow you to stop the loop early. If you use a for loop, you can use `break` after the `remediationAggregation` is set to `inform/enforce (overidden)` since no other value can be set.

Reply:
For efficiency since after `remediationAggregation` is set to `inform/enforce (overidden)` no other value can possibly be set.

Reply:
Oh I see!! ok I will change to `for`